src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java

```diff
@@ -65,16 +67,21 @@ public static org.apache.parquet.schema.Type parseIcebergDataTypeStringToParquet
       int id,
       String name) {
     Type icebergType = deserializeIcebergType(icebergDataType);
```

Comment:
Lets discuss this tomorrow, had an idea that we can explore after this PR goes in / as a future enhancement to reduce complexity :

Do we really need to use parquet types in iceberg ingestion, when there is already an iceberg data type hierarchy that works well?
What if we added an IcebergRowBuffer just like ParquetRowBuffer. IcebergRowBuffer will use iceberg data types, and ParquetRowBuffer will use parquet data types (for BDEC).

Reply:
The Iceberg types does not have any information about the actual physical type. As long as we are writing parquet I think we always need parquet type.

Reply:
i see, ok.